src/app/components/dashboard/dashboard.component.html

```diff
@@ -1,241 +1,277 @@
-
-                        </td>
-                        <td></td>
-                    </tr>
-                </ng-template>
-
-
-                <ng-template pTemplate="rowexpansion" let-worker>
-                    <tr>
-                        <td></td>
-                        <td><p-skeleton></p-skeleton></td>
-                        <td><p-skeleton></p-skeleton></td>
-                        <td><p-skeleton></p-skeleton></td>
-                        <td><p-skeleton></p-skeleton></td>
-                        <td><p-skeleton></p-skeleton></td>
-                    </tr>
-                </ng-template>
-
-
-            </p-table>
+          </div>
+          <span class="text-green-500 font-medium"
+            >Weight:
+            <ng-container *ngIf="data.networkInfo; else loading">
+              {{ data.networkInfo.currentblockweight | number }}</ng-container
+            >
+            <ng-template #loading><p-skeleton></p-skeleton></ng-template>
+          </span>
+        </div>
+      </div>
+    </ng-container>
+  </div>
+  <div class="card">
+    <ng-container *ngIf="data.clientInfo; else loadingTable">
+      <p-table
+        #dataTable
+        [rowHover]="true"
+        groupRowsBy="name"
+        dataKey="name"
+        rowGroupMode="subheader"
+        [value]="data.clientInfo.workers"
+        [expandedRowKeys]="expandedRows$ | async"
+        [tableStyle]="{ 'min-width': '50rem' }"
+      >
+        <ng-template pTemplate="header">
+          <tr>
+            <th>Name</th>
+            <th>Session ID</th>
+            <th>Hash Rate</th>
+            <th>Session Best Difficulty</th>
+            <th>Uptime</th>
+            <th>Last Seen</th>
+          </tr>
         </ng-template>
-
-    </div>
-
-
-    <div class="card chart">
-        <ng-container *ngIf="chartData$ | async as chartData; else loadingChart">
-            <p-chart [data]="chartData" [options]="chartOptions"></p-chart>
-        </ng-container>
-        <ng-template #loadingChart>
-            <p-skeleton width="100%" height="40vh"></p-skeleton>
+        <ng-template
+          pTemplate="groupheader"
+          let-worker
+          let-rowIndex="rowIndex"
+          let-expanded="expanded"
+        >
+          <tr [routerLink]="[worker.name]">
+            <td>
+              <button
+                (click)="$event.stopImmediatePropagation()"
+                type="button"
+                pButton
+                pRipple
+                [pRowToggler]="worker"
+                class="p-button-text p-button-rounded p-button-plain mr-2"
+                [icon]="expanded ? 'pi pi-chevron-down' : 'pi pi-chevron-right'"
+              ></button>
+
+              <span class="worker-name font-bold ml-2">{{ worker.name }} </span>
+            </td>
+            <td>
+              {{
+                getSessionCount(worker.name, data.clientInfo.workers)
+              }}
+              Sessions
+            </td>
+            <td>
+              {{
```

Comment:
Not a fan of using prettier in html markup - makes things like this look really ugly